icu4c/source/common/locid.cpp

```diff
@@ -626,19 +626,6 @@ Locale& Locale::init(const char* localeID, UBool canonicalize)
             variantBegin = (int32_t)(field[variantField] - fullName);
         }
 
-        if (length == 4 && uprv_stricmp(fullName, "root") == 0) {
```

Comment:
I am surprised that you can just delete the root/und handling from Locale::init().
What then maps those to ""?